cni/pkg/repair/netns.go

```diff
@@ -33,46 +31,48 @@ func getPidNamespace(pid int) string {
 	return "/host/proc/" + strconv.Itoa(pid) + "/ns/net"
 }
 
-func runInHost[T any](f func() (T, error)) (T, error) {
+func runInNS[T any](pidNs string, f func() (T, error)) (T, error) {
 	var res T
-	ns, err := netns.GetNS(getPidNamespace(1))
+	ns, err := netns.GetNS(pidNs)
 	if err != nil {
-		return res, fmt.Errorf("failed to get host network: %v", err)
+		return res, fmt.Errorf("failed to get network namespace %v: %v", pidNs, err)
 	}
 	err = ns.Do(func(ns netns.NetNS) error {
 		var err error
 		res, err = f()
 		return err
 	})
 	if err != nil {
-		return res, fmt.Errorf("in host network: %v", err)
+		return res, fmt.Errorf("in network namespace %v: %v", ns, err)
 	}
 	return res, nil
 }
 
-func findNetworkIDByIP(ip string) (int, error) {
-	link, err := getLinkWithDestinationOf(ip)
-	if err != nil {
-		return 0, fmt.Errorf("find link for %v: %v", ip, err)
-	}
-	return link.Attrs().NetNsID, nil
+func runInHost[T any](f func() (T, error)) (T, error) {
+	return runInNS(getPidNamespace(1), f)
 }
 
-func getLinkWithDestinationOf(ip string) (netlink.Link, error) {
-	routes, err := netlink.RouteListFiltered(
-		netlink.FAMILY_V4,
-		&netlink.Route{Dst: &net.IPNet{IP: net.ParseIP(ip), Mask: net.CIDRMask(32, 32)}},
-		netlink.RT_FILTER_DST)
-	if err != nil {
-		return nil, err
+func checkInterfacesForMatchingAddr(targetAddr string) (match bool, err error) {
+	var interfaces []net.Interface
+	if interfaces, err = net.Interfaces(); err != nil {
+		return false, fmt.Errorf("failed to get interfaces")
 	}
 
-	if len(routes) == 0 {
-		return nil, fmt.Errorf("no routes found for %s", ip)
+	for _, ief := range interfaces {
+		var addrs []net.Addr
+		if addrs, err = ief.Addrs(); err != nil {
+			return
+		}
+		for _, addr := range addrs {
+			switch v := addr.(type) {
+			case *net.IPNet:
+				if v.IP.String() == targetAddr {
```

Comment:
Will this work with 4in6 addresses? do we need to https://pkg.go.dev/net/netip#Addr.Unmap or similar first?